GameVersion.cpp

```diff
@@ -347,3 +348,170 @@ bool GameVersion::isLargerishThan(const GameVersion& rhs) const
 
 	return true;
 }
+
+std::optional<GameVersion> GameVersion::extractLauncherVersion(const std::string& filePath)
```

Comment:
Have you considered using a JSON library? I can recommend nlhomann one, comes in a single header form.